tdm-admin/src/app/page/Setting.tsx

```diff
@@ -0,0 +1,158 @@
+                direction="row"
+                justifyContent="center"
+                alignItems="flex-start"
+                spacing={2}
+            >
+                <Paper className="setting-block" elevation={0}>
+                    <Skeleton animation="wave" variant="rounded" height={500} />
+                </Paper>
+                <Paper className="setting-block" elevation={0}>
+                    <Skeleton animation="wave" variant="rounded" height={500} />
+                </Paper>
+            </Stack>
+        </Box>
+    );
+};
+
+const Setting = () => {
+    const [localConfig, setLocalConfig] = useState<Config | undefined>(undefined);
+    const [configUpdateIndex, setConfigUpdateIndex] = useState<number>(0);
+    const [asChange, setAsChange] = useState<boolean>(false);
+    const [snackbar, setSnackbar] = useState({ open: false, message: '' });
+
+    const { data, isLoading, isFetching } = useQuery({
+        queryKey: ['config', configUpdateIndex],
+        queryFn: () => {
+            return config();
+        },
+    });
+
+    useEffect(() => {
+        setLocalConfig(cloneDeep(data));
+        setAsChange(false);
+    }, [data, configUpdateIndex]);
+
+    useEffect(() => {
+        setAsChange(!isEqual(data, localConfig));
+    }, [data, localConfig]);
+
+    const handleSnackBarClose = () => {
+        setSnackbar({ open: false, message: '' });
+    };
+
+    const handleWrapperChange = (wrappers: ConfigWrapper[]) => {
+        if (localConfig) {
+            const tmpConfig: Config = cloneDeep(localConfig);
+            tmpConfig.wrappers = wrappers;
+            setLocalConfig(tmpConfig);
+        }
+    };
+
+    const handleEnrichmentChange = (enrichments: ConfigEnrichment[]) => {
+        if (localConfig) {
+            const tmpConfig: Config = cloneDeep(localConfig);
+            tmpConfig.enrichments = enrichments;
+            setLocalConfig(tmpConfig);
+        }
+    };
+
+    const handleCancelClick = () => {
+        setConfigUpdateIndex(configUpdateIndex + 1);
+    };
+
+    const handleSaveClick = () => {
+        if (localConfig) {
+            configPost(localConfig).then((res) => {
+                if (res.ok) {
+                    setSnackbar({
+                        open: true,
+                        message: 'La configuration a été sauvegardé',
+                    });
+                    setConfigUpdateIndex(configUpdateIndex + 1);
+                } else {
+                    setSnackbar({
+                        open: true,
+                        message: 'Un erreur durrent la sauvegardé a été rencontrais',
+                    });
+                }
+            });
+        }
+    };
+
+    if (isLoading || isFetching) {
+        return <SettingSkeleton />;
+    }
+
+    if (!data || !localConfig) {
+        return (
+            <Box>
+                <Alert severity="error">Impossible de récupérais la configuration.</Alert>
```

Comment:
<!-- Typo comments are like todo:, where the main issue is a mispelling.  -->
```suggestion
                <Alert severity="error">Impossible de récupérer la configuration.</Alert>
```
**typo:** ‏